core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -537,6 +539,60 @@ public T proxyDetector(ProxyDetector proxyDetector) {
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Provides a service config to the channel. The channel will use the default service config when
+   * the name resolver provides no service config or if the channel disables lookup service config
+   * from name resolver (see {@link #lookUpServiceConfig(boolean)}). The argument
+   * {@code serviceConfig} is a nested map representing a Json object in the most natural way:
+   *
+   *        <table border="1">
+   *          <tr>
+   *            <td>Json entry</td><td>Java Type</td>
+   *          </tr>
+   *          <tr>
+   *            <td>object</td><td>{@link Map}</td>
```

Comment:
`Map<String, ?>`

Reply:
Generic types, including wildcards sometimes are very inconvenient to use. I would let the users be flexible to choose whatever generic type parameter that's most convenient for them, (either the users want to minimize the times to cast and suppress unchecked, or want to minimize errorprone at compile time), as long as it follows the table as a whole.